projects/ag-grid/documentation/code-examples/data-entry-grid/basic/data-entry-grid-docs-demo-edit-modal.component.ts

```diff
@@ -0,0 +1,163 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+} from '@angular/core';
+import {
+  SkyAgGridService,
+  SkyAutocompleteProperties,
+  SkyCellType,
+  SkyDatepickerProperties,
+} from '@skyux/ag-grid';
+import { SkyAutocompleteSelectionChange } from '@skyux/lookup';
+import { SkyModalInstance } from '@skyux/modals';
+
+import {
+  CellValueChangedEvent,
+  ColDef,
+  GridApi,
+  GridOptions,
+  GridReadyEvent,
+  ICellEditorParams,
+  NewValueParams,
+  RowNode,
+} from 'ag-grid-community';
+
+import {
+  SkyAgGridDemoRow,
+  SKY_DEPARTMENTS,
+  SKY_JOB_TITLES,
+} from './data-entry-grid-docs-demo-data';
+import { SkyDataEntryGridEditModalContext } from './data-entry-grid-docs-demo-edit-modal-context';
+
+@Component({
+  selector: 'app-data-entry-grid-docs-demo-edit-modal',
+  templateUrl: './data-entry-grid-docs-demo-edit-modal.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SkyDataEntryGridEditModalComponent {
+  public columnDefs: ColDef[];
+  private gridApi: GridApi | undefined;
+  public gridData: SkyAgGridDemoRow[];
+  public gridOptions: GridOptions;
+
+  constructor(
+    private agGridService: SkyAgGridService,
+    public context: SkyDataEntryGridEditModalContext,
+    public instance: SkyModalInstance,
+    private changeDetector: ChangeDetectorRef
+  ) {
+    this.columnDefs = [
+      {
+        field: 'name',
+        headerName: 'Name',
+      },
+      {
+        field: 'age',
+        headerName: 'Age',
+        type: SkyCellType.Number,
+        maxWidth: 100,
+        editable: true,
+      },
+      {
+        field: 'startDate',
+        headerName: 'Start Date',
```

Comment:
Our UX content generally follows sentence-style capitalization (i.e. only capitalizing the first word and proper nouns), so if this header appears in the UI, assume we should lowercase the "D" for consistency.
```suggestion
        headerName: 'Start date',
```